theme-utils.mjs

```diff
@@ -1130,3 +1136,106 @@ export async function executeCommand(command, logResponse) {
 		});
 	});
 }
+
+async function escapePatterns() {
+	// get staged files
+	const staged = (await executeCommand(`git diff --cached --name-only`)).split('\n');
+	// get unstaged, untracked files
+	const unstaged = (await executeCommand(`git ls-files -m -o --exclude-standard`)).split('\n');
+	
+	// avoid duplicates and filter pattern files
+	const patterns = [...new Set([...staged, ...unstaged])].filter(file => file.match(/.*\/patterns\/(\w+).php/g));
+
+	// arrange patterns by theme
+	const themePatterns = patterns.reduce((acc, file) => {
+		const themeSlug = file.split('/').shift();
+		return {
+			...acc,
+			[themeSlug]: (acc[themeSlug] || []).concat(file)
+		};
+	}, {});
+
+	Object.entries(themePatterns).forEach(async ([themeSlug, patterns]) => {
+		console.log(getPatternTable(themeSlug, patterns));
+	
+		const prompt = await inquirer.prompt([{
+			type: 'input',
+			message: 'Verify the theme slug',
+			name: "themeSlug",
+			default: themeSlug
+		}]);
+
+		if (!prompt.themeSlug) {
+			return;
+		}
+
+		const rewriter = getReWriter(prompt.themeSlug);
+		patterns.forEach(file => {
+			const tmpFile = `${file}-tmp`;
+			const rstream = fs.createReadStream( file, { encoding: 'UTF-8' } );
+			const wstream = fs.createWriteStream( tmpFile, { encoding: 'UTF-8' } );
+			wstream.on('finish', () => {
+				fs.renameSync(tmpFile, file);
+			});
+
+			rstream.pipe(rewriter).pipe(wstream);
+		});
+	});
+
+
+	// Helper functions
+	function getReWriter(themeSlug) {
+		const rewriter = new RewritingStream();
+
+		rewriter.on('text', (_, raw) => {
+			rewriter.emitRaw(escapeText(raw, themeSlug));
+		});
+
+		rewriter.on('startTag', (startTag) => {
+			if (startTag.tagName === 'img') {
+				const attrs = startTag.attrs.filter(attr => ['src', 'alt'].includes(attr.name));
+				attrs.forEach(attr => {
+					if (attr.name === 'src') {
+						attr.value = escapeImagePath(attr.value);
+					} else if (attr.name === 'alt') {
+						attr.value = escapeText(attr.value, themeSlug, true);
+					}
+				});
+			}
+
+			rewriter.emitStartTag(startTag);
+		});
+
+		return rewriter;
+	}
+
+	function escapeText(text, themeSlug, isAttr = false) {
+		const trimmedText = text && text.trim();
+		if (!themeSlug || !trimmedText || trimmedText.startsWith(`<?php`)) return text;
+		const escFunction = isAttr ? 'esc_attr__' : 'esc_html__';
+		return `<?php echo ${escFunction}( '${text.replace('\'', '\\\'')}', '${themeSlug}' ); ?>`;
+	}
+
+	function escapeImagePath(src) {
+		if (!src || src.trim().startsWith('<?php')) return src;
+		
+		const assetsDir = 'assets';
+		const parts = src.split('/');
+		const resultSrc = parts.slice(parts.indexOf(assetsDir)).join('/');
+		return `<?php echo esc_url( get_template_directory_uri() ); ?>/${resultSrc}`;
```

Comment:
It may be more useful to prepend the usual path where we place static image files: 
```suggestion
		return `<?php echo esc_url( get_template_directory_uri() ); ?>/assets/images/${resultSrc}`;
```

Reply:
`resultSrc` is calculated to include `assets` and the image path in the previous line.